tools/emacs/soutei-mode.el

```diff
@@ -0,0 +1,49 @@
+(defvar soutei-mode-hook nil)
+
+(defface soutei-keyword-face
+  '((t (:inherit font-lock-keyword-face)))
+  "Soutei keywords")
+
+(defface soutei-operator-face
+  '((t (:inherit font-lock-builtin-face)))
+  "Soutei builtin operators")
+
+(defvar soutei-indent-width prolog-indent-width
+  "indentation width used by auto-indent functions")
+
+(defvar soutei-mode-map (make-keymap)
+  "Keymap for soutei major mode")
+
+(defvar soutei-mode-syntax-table
+  (let ((synTable (make-syntax-table)))
+    (modify-syntax-entry ?- "_" synTable)
+    (modify-syntax-entry ?\; "<" synTable)
```

Comment:
I'm not sure whether here or elsewhere is where it happens, but `%` is getting treated as a comment character when I try this locally.

Reply:
`prolog-mode` sets `%` as a comment character, and we inherit their syntax table.  Fixed in the second commit.